src/WebJobs.Script.WebHost/WebScriptHostManager.cs

```diff
@@ -35,7 +35,13 @@ public WebScriptHostManager(ScriptHostConfiguration config) : base(config)
             {
                 { triggerParameter.Name, request }
             };
-            await Instance.CallAsync(function.Name, arguments, cancellationToken);
+
+            // Suspend the current synchronization context so we don't pass the ASP.NET
+            // context down to the function.
+            using (var syncContextSuspensionScope = new SuspendedSynchronizationContextScope())
+            {
+                await Instance.CallAsync(function.Name, arguments, cancellationToken);
+            }
```

Comment:
This removes the `SynchronizationContext` when invoking the function, preventing it from flowing and from being captured by Tasks/continuations downstream.